### PR TITLE
c++ msm uses new API

### PR DESCRIPTION
**Pull request by svpolonsky**
## Describe the changes

This PR allows c++ MSM example to compile with new API

## Linked Issues

Resolves #

### CODE REVIEW

File: examples/c++/msm/example.cu [hunk 1]
```diff
@@ -1,59 +1,125 @@
...
+  // Execute the MSM kernel
+  msm::MSM<scalar_t, affine_t, projective_t>(scalars, points, msm_size, config, &result);
+  // Record the stop event on the stream
+  cudaEventRecord(stop, stream);
+  // Wait for the stop event to complete
+  cudaEventSynchronize(stop);
+  // Calculate the elapsed time between the start and stop events
+  cudaEventElapsedTime(&time, start, stop);
+  // Destroy the events
+  cudaEventDestroy(start);
```
Comment (DmytroTym) [inline examples/c++/msm/example.cu:74]: Small note: possible to avoid destroying events here and recreating them on lines 98-99

**Review by DmytroTym — COMMENTED** [review]
I wasn't immediately able to build the example, but running `export CPATH=path_to_icicle_root` helped. Not sure if it's possible to somehow know where the root is from a shell script, if not I guess worth mentioning this in README?
In general no critical issues with the PR, just some stylistic comments.
Please rebase the PR onto `dev`, we agreed to use it for now.

File: examples/c++/msm/README.md [hunk 2]
```diff
@@ -6,44 +6,41 @@ We recommend to run our examples in [ZK-containers](../../ZK-containers.md) to s
...
 
 ```c++
-#include "appUtils/msm/msm.cu"           // template
-#include "curves/bn254/curve_config.cuh" // curve
-using namespace BN254;
+#define CURVE_ID 1
+#include "icicle/appUtils/msm/msm.cu"
 ...
-large_msm<scalar_t, projective_t, affine_t>(scalars,points,size,result,on_device,big_triangle,bucket_factor,stream)
+msm::MSMConfig config = {...};
```
[the remark this answers is not in the document]
Comment (svpolonsky) [reply examples/c++/msm/README.md:22]: Now I use `msm::DefaultMSMConfig()`.  I have to use `msm.cu` to instantiate the templates.

File: examples/c++/msm/example.cu [hunk 3]
```diff
@@ -1,59 +1,125 @@
...
+    0,     // bitsize
+    10,    // large_bucket_factor
+    batch_size,    
+    false, // are_scalars_on_device
+    false, // are_scalars_montgomery_form
+    false, // are_points_on_device
+    false, // are_points_montgomery_form
+    false,  // are_results_on_device
+    false, // is_big_triangle
+    true,  // is_async
```
[the remark this answers is not in the document]
Comment (svpolonsky) [reply examples/c++/msm/example.cu]: now using the default

File: examples/c++/msm/example.cu [hunk 4]
```diff
@@ -1,59 +1,125 @@
...
+// select the curve. TODO: can we use curve names instead of IDs?
+#define CURVE_ID 1
 
 // include MSM template
-#include "appUtils/msm/msm.cu"
-// select the curve
-#include "curves/bn254/curve_config.cuh"
-
-using namespace BN254;
+#include "icicle/appUtils/msm/msm.cu"
```
[the remark this answers is not in the document]
Comment (svpolonsky) [reply examples/c++/msm/example.cu:9]: I need .cu for templates

**Pull request merged**